esp-hal/src/spi/master.rs

```diff
@@ -2476,9 +2478,15 @@ pub trait Instance: private::Sealed {
 
     fn cs_signal(&self) -> OutputSignal;
 
-    fn enable_peripheral(&self);
+    #[inline(always)]
+    fn enable_peripheral(&self) {
+        PeripheralClockControl::enable(self.peripheral());
+    }
 
-    fn reset_peripheral(&self);
+    #[inline(always)]
+    fn reset_peripheral(&self) {
+        PeripheralClockControl::reset(self.peripheral());
+    }
```

Comment:
Any reason not to inline this at the call site?

Reply:
Smaller diff, consistency, and possible future ideas, pick a few :)

There is a chance we can allow accessing SPI1 and we'd probably have to allow overriding this then.